core/controllers/reader_test.py

```diff
@@ -362,3 +363,166 @@ def test_ratings_by_different_users(self):
+            'newuser has submitted a new report on the exploration Title'
+            ' on the grounds of AD .\n'
+            'You can modify the exploration by clicking "Edit Title".\n'
+            '\n'
+            'Thanks!\n'
+            '- The Oppia Team\n'
+            '\n'
+            'You can change your email preferences via the Preferences page.')
+
+        with self.can_send_emails_ctx:
+
+            self.process_and_flush_pending_tasks()
+
+            messages = self.mail_stub.get_sent_messages(to=self.MODERATOR_EMAIL)
+            self.assertEqual(len(messages), 1)
+            self.assertEqual(
+                messages[0].html.decode(),
+                expected_email_html_body)
+            self.assertEqual(
+                messages[0].body.decode(),
+                expected_email_text_body)"""
+
+
+class FlagExplorationEmailHandlerTest(test_utils.GenericTestBase):
+
+    """def setUp(self):
+        super(FlagExplorationEmailHandlerTest, self).setUp()
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.editor_id = self.get_user_id_from_email(self.EDITOR_EMAIL)
+
+        self.signup(self.NEW_USER_EMAIL, self.NEW_USER_USERNAME)
+        self.new_user_id = self.get_user_id_from_email(self.NEW_USER_EMAIL)
+
+        self.signup(self.MODERATOR_EMAIL, self.MODERATOR_USERNAME)
+        self.moderator_id = self.get_user_id_from_email(self.MODERATOR_EMAIL)
+        self.set_moderators([self.MODERATOR_USERNAME])
+
+        self.exploration = self.save_new_default_exploration(
+            'A', self.editor_id, 'Title')
+        self.can_send_emails_ctx = self.swap(
+            feconf, 'CAN_SEND_EMAILS', True)
+        self.report_text = 'AD'
+
+    def test_that_emails_are_sent(self):
+        expected_email_html_body = (
+            'Hello Moderator,<br>'
+            'newuser has submitted a new report on the exploration Title'
+            ' on the grounds of AD .<br>'
+            'You can modify the exploration by clicking '
+            '<a href="https://www.oppia.org/create/A">'
+            '"Edit Title"</a>.<br>'
+            '<br>'
+            'Thanks!<br>'
+            '- The Oppia Team<br>'
+            '<br>'
+            'You can change your email preferences via the '
+            '<a href="https://www.example.com">Preferences</a> page.')
+
+        expected_email_text_body = (
+            'Hello Moderator,\n'
+            'newuser has submitted a new report on the exploration Title'
+            ' on the grounds of AD .\n'
+            'You can modify the exploration by clicking "Edit Title".\n'
+            '\n'
+            'Thanks!\n'
+            '- The Oppia Team\n'
+            '\n'
+            'You can change your email preferences via the Preferences page.')
+
+        with self.can_send_emails_ctx:
+            moderator_services.enqueue_flag_exploration_email_task(
+                self.exploration.id, self.report_text)
+
+            self.process_and_flush_pending_tasks()
```

Comment:
Thanks @seanlip , I figured out the issue. It was that a user has to be logged in while reporting issues. I updated the tests and it works now.